### PR TITLE
Add time unit to lease length configuration

### PR DESCRIPTION
**Pull request by ChrisGreenaway**

### CODE REVIEW

File: lease/entity-server/src/main/resources/lease-service.xsd [hunk 1]
```diff
@@ -17,28 +17,53 @@
...
+          </xs:documentation>
+        </xs:annotation>
+      </xs:element>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="lease-length-type">
+    <xs:simpleContent>
+      <xs:extension base="xs:positiveInteger">
+        <xs:attribute name="unit" type="lease:time-unit" use="required">
```
Comment (ljacomet) [inline lease/entity-server/src/main/resources/lease-service.xsd]: I would go with optional having a default, let's say `milliseconds`

File: lease/entity-server/src/main/resources/lease-service.xsd [hunk 2]
```diff
@@ -17,28 +17,53 @@
...
       </xs:documentation>
     </xs:annotation>
-    <xs:complexType>
-      <xs:sequence>
-        <xs:element type="xs:positiveInteger" name="lease-length"  minOccurs="1" maxOccurs="1">
+  </xs:element>
+
+  <xs:complexType name="connection-leasing-type">
+    <xs:sequence>
+      <xs:element name="lease-length" type="lease:lease-length-type" minOccurs="1" maxOccurs="1">
```
Comment (ljacomet) [inline lease/entity-server/src/main/resources/lease-service.xsd]: Still find `lease-length` less than ideal. What about `lease-duration`?

File: lease/entity-server/src/main/resources/lease-service.xsd (hunk 1 above)
Comment (lorban) [inline lease/entity-server/src/main/resources/lease-service.xsd]: I'd rather use `seconds` as a default.

Comment (ChrisGreenaway) [inline lease/entity-server/src/main/resources/lease-service.xsd]: I don't think there should be a default.

File: lease/entity-server/src/main/resources/lease-service.xsd (hunk 2 above)
Comment (ChrisGreenaway) [inline lease/entity-server/src/main/resources/lease-service.xsd]: The word "length" really goes with "lease". If you were renting a house, the conversation would be how long a lease do you want, not what duration do you want on your lease.

Comment (ljacomet) [inline lease/entity-server/src/main/resources/lease-service.xsd]: Ok, respect the native speaker there 😉

File: lease/entity-server/src/main/resources/lease-service.xsd (hunk 1 above)
Comment (ljacomet) [inline lease/entity-server/src/main/resources/lease-service.xsd]: Let's go with no default, which enables adding one later if needs be 😉

**Comment by chrisdennis** [comment]
Am I reading this right that parsing a lease configuration logs two things at info, neither of which are the actual lease length?  Whats the reasoning here?

**Pull request merged**